web/app/src/components/Settings.vue

```diff
@@ -48,15 +57,15 @@ export default {
       this.setRefreshInterval(this.$refs.refreshInterval.value);
     },
     toggleDarkMode() {
-      if (localStorage.theme === 'dark') {
-        localStorage.theme = 'light';
+      if (getTheme() === 'dark') {
+        document.cookie = `theme=light; path=/; max-age=31536000; samesite=strict`;
       } else {
-        localStorage.theme = 'dark';
+        document.cookie = `theme=dark; path=/; max-age=31536000; samesite=strict`;
```

Comment:
This will likely cause the first toggle to do nothing if you don't already have a cookie:

1. User loads page (no `theme` cookie)
2. Defaults to dark theme because `wantsDarkMode()` returns `true` if there's no theme
3. User clicks on theme (moon/sun) button
4. Because `getTheme()` doesn't return `dark`, it falls on the `else` block and a cookie with `theme=dark` is created, which means that the first click doesn't do anything
5. Second click will change from dark -> light because the cookie now exists

I'll push some code to fix this as well.

Reply:
cb3ebb7 should address this, but I don't have time to test this in-depth right now. Would you mind giving it a shot and letting me know? 🙏